src/test/java/com/team4/testingsystem/controllers/CoachGradeControllerTest.java

```diff
@@ -0,0 +1,162 @@
+package com.team4.testingsystem.controllers;
+
+import com.team4.testingsystem.dto.CoachGradeDTO;
+import com.team4.testingsystem.entities.CoachGrade;
+import com.team4.testingsystem.entities.Question;
+import com.team4.testingsystem.exceptions.CoachGradeAlreadyExistsException;
+import com.team4.testingsystem.exceptions.GradeNotFoundException;
+import com.team4.testingsystem.exceptions.QuestionNotFoundException;
+import com.team4.testingsystem.exceptions.TestNotFoundException;
+import com.team4.testingsystem.services.CoachGradeService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import org.junit.jupiter.api.Assertions;
+
+import java.util.Collections;
+import java.util.List;
+
+@ExtendWith(MockitoExtension.class)
+class CoachGradeControllerTest {
+    @Mock
+    private CoachGradeService gradeService;
+
+    @InjectMocks
+    private CoachGradeController gradeController;
+
+    @Mock
+    private CoachGrade coachGrade;
+
+    @Mock
+    private com.team4.testingsystem.entities.Test test;
+
+    @Mock
+    private Question question;
+
+    private final Long testId = 1L;
+    private final Long questionId = 1L;
```

Comment:
Let's use different ids for questions, tests, etc. Collision here may cause some passing tests even though they have errors